public/app/plugins/datasource/loki/components/LokiExploreQueryEditor.tsx

```diff
@@ -0,0 +1,110 @@
+// Libraries
+import React, { memo, useState, useEffect } from 'react';
+import _ from 'lodash';
+
+// Types
+import { AbsoluteTimeRange, ExploreQueryFieldProps, ExploreMode } from '@grafana/data';
+import { LokiDatasource } from '../datasource';
+import { LokiQuery, LokiOptions } from '../types';
+import { LokiQueryField } from './LokiQueryField';
+import { useLokiSyntax } from './useLokiSyntax';
+import { LokiExploreExtraField } from './LokiExploreExtraField';
+
+type Props = ExploreQueryFieldProps<LokiDatasource, LokiQuery, LokiOptions>;
+
+export const LokiQueryEditor = memo(function LokiQueryEditor(props: Props) {
+  const { query, data, datasource, exploreMode, history, onChange, onRunQuery } = props;
+
+  let absolute: AbsoluteTimeRange;
+  if (data && !_.isEmpty(data.request)) {
+    const { range } = data.request;
+
+    absolute = {
+      from: range.from.valueOf(),
+      to: range.to.valueOf(),
+    };
+  } else {
+    absolute = {
+      from: Date.now() - 10000,
+      to: Date.now(),
+    };
+  }
+
+  const [maxLines, setMaxLines] = useState(query?.maxLines?.toString() || datasource?.maxLines?.toString());
+
+  const { isSyntaxReady, setActiveOption, refreshLabels, ...syntaxProps } = useLokiSyntax(
+    datasource.languageProvider,
+    absolute
+  );
+
+  function onChangeQueryLimit(value: string, override?: boolean) {
+    const { query, onChange, onRunQuery } = props;
+    if (onChange) {
+      const nextQuery = { ...query, maxLines: preprocessMaxLines(value) };
+      onChange(nextQuery);
+
+      if (override && onRunQuery) {
+        onRunQuery();
+      }
+    }
+  }
+
+  function preprocessMaxLines(value: string): number {
```

Comment:
Did this change? It seems like it works still the same way, ie `-1` or `aaa` will result in `No logs found.Scan for older logs`

Reply:
@aocenas There's a [type property on the input field](https://github.com/grafana/grafana/pull/21497/files#diff-c560b869d69182f7a1ff79a0eac8325fR23) initialized there to [a number](https://github.com/grafana/grafana/pull/21497/files#diff-71cc63d2d8d34b4d19ac1b1e809373c8R100), so there shouldn't be any issues with non-numerical values, also added an additional check in [`onReturnKeyDown` method](https://github.com/grafana/grafana/pull/21497/files#diff-71cc63d2d8d34b4d19ac1b1e809373c8R75) to prevent query from running when the value of input field is <= 0.

Reply:
I mean I tried it right now and seems like it allows me to input negative values or non numbers:
<img width="1602" alt="Screenshot 2020-01-15 at 11 58 26" src="https://user-images.githubusercontent.com/1014802/72428663-78375f80-378e-11ea-9498-3414c279dbaf.png">

Reply:
Hmm ok it does not allow for non numeric values (except for `e` because of exponent notation) but negative values still seem to work as I mentioned.

Reply:
When fixed can we also add a test for this component?

Reply:
Can you write down here what did you decide with Diana regarding how this should function?

Reply:
For negative inputs we decided to fall back to 0, when we implement new forms we'll display input validation errors.